tools/ddprobe.cpp

```diff
@@ -92,24 +214,41 @@ test_dxgi_duplication(dxgi::adapter_t &adapter, dxgi::output_t &output) {
     &device,
     nullptr,
     nullptr);
+
   if (FAILED(status)) {
-    std::cout << "Failed to create D3D11 device for DD test [0x"sv << util::hex(status).to_string_view() << ']' << std::endl;
+    std::cout << "Failed to create D3D11 device for DD test [0x" << std::hex << status << "]" << std::endl;
     return status;
   }
 
   dxgi::output1_t output1;
   status = output->QueryInterface(IID_IDXGIOutput1, (void **) &output1);
   if (FAILED(status)) {
-    std::cout << "Failed to query IDXGIOutput1 from the output"sv << std::endl;
+    std::cout << "Failed to query IDXGIOutput1 from the output [0x" << std::hex << status << "]" << std::endl;
     return status;
   }
 
   // Ensure we can duplicate the current display
   syncThreadDesktop();
 
-  // Return the result of DuplicateOutput() to Sunshine
+  // Attempt to duplicate the output
   dxgi::dup_t dup;
-  return output1->DuplicateOutput((IUnknown *) device.get(), &dup);
+  HRESULT result = output1->DuplicateOutput(static_cast<IUnknown *>(device.get()), &dup);
+
+  if (SUCCEEDED(result)) {
+    // If duplication is successful, test frame capture
+    HRESULT captureResult = test_frame_capture(dup, device.get());
+    if (SUCCEEDED(captureResult)) {
+      return S_OK;
+    }
+    else {
```

Comment:
Redundant else.

Reply:
I prefer to call that explicitly defined else 😅


That's just a matter of preference basically, even if it doesn't matter a lot of people prefer seeing else as it is easier to read.

Reply:
If you don't mind... for this code base I prefer to remove the redundancy. It reduces nesting and is easier to read (IMO).

Reply:
As we don't have a coding guideline, I pointed this out, we discussed it and that's that.

Now, how do I resolve this...

Reply:
> If you don't mind... for this code base I prefer to remove the redundancy. It reduces nesting and is easier to read (IMO).

I honestly don't see how it's easier to read by removing braces and hiding the branch path by using a return

By removing the else, you make it harder for IDE to be able to simply hide the code path by minimizing it.

If anything it makes it harder to maintain in my opinion, but whatever, the debate of if/else is one of the most controversial ones for developers, I'll remove it

Reply:
JetBrains IDEs complain about redundant else clauses, by default.

![image](https://github.com/user-attachments/assets/9e7442be-730c-449a-86ed-5780f37c507d)

Reply:
> JetBrains IDEs complain about redundant else clauses, by default.
> 
> ![image](https://private-user-images.githubusercontent.com/42013603/356899439-9e7442be-730c-449a-86ed-5780f37c507d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM0MDc2NzYsIm5iZiI6MTcyMzQwNzM3NiwicGF0aCI6Ii80MjAxMzYwMy8zNTY4OTk0MzktOWU3NDQyYmUtNzMwYy00NDlhLTg2ZWQtNTc4MGYzN2M1MDdkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODExVDIwMTYxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE2MWVmMDgzNjQyYWNlNjRmMjMyMjRhMjFhYjgwZjlkOTg2N2JlNjk0NjdiMjBhZDNlZjU0NzVlMjFiNDgyMDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5w1iRIpehxR_VUGgDkyJD4fE8RtLHS1IKy3ZusPMNzk)
Yeah I am just going to say we agree to disagree there.

Imagine if, when you're trying to plan an event, you thought, "If event_needs_planning... return plan_event; return event_planned!" But that’s not how most humans think. Instead, we think more like, "If event_needs_planning, plan_event; otherwirse I don't need to plan anything."

The "else" statement mirrors how we naturally think, which is why I argue it's easier to understand and follow the flow. It also makes it much easier to hide branches in the IDE, which is the main benefit of it. If you followed every rule we would end up with complete garbage code like https://github.com/EnterpriseQualityCoding/FizzBuzzEnterpriseEdition

Reply:
Oh I just reread that jetbrains code issue and it's talking about an entirely different issue.

It warns when you have a completely redundant if/else statement that can be simplified to a single return, without needing an if or else statement entirely.

That's not the same thing as what we were talking about earlier.

Either way all of those are "fixed" now